### PR TITLE
Add Custom Component Feature to ExtensibleForm.

### LINKED ISSUE #13785: The additionally added mandatory propertie is functionally different from other mandatory fields and  localization seems broken.
**App:** Angular

- The additionally added mandatory propertie is functionally different from other mandatory fields

image

- The localization of the additionally added property on the User page appears broken.
image

### PR DESCRIPTION
**Pull request by mahmut-gundogdu**
Resolves #13785

Test Preparation
 - go to npm>ng_packs and run `yarn run build:all`
- run backend with extra-properties 
- run ng-packs/dev-app. 

Test steps
- [x] my-profile page re-written with extensible form. So it should work as before.  All functionality that in my-profile should be tested.
- [x] Add extra properties  on Backend (https://docs.abp.io/en/abp/latest/Module-Entity-Extensions)
- [x] The extra properties should visible and changeable on the profile page. 

the task related to https://github.com/abpframework/abp/pull/13700

### CODE REVIEW

**Review by muhammedaltug — APPROVED** [review]

**Comment by gizemmutukurt** [comment]
I requested some fixes and changes, you can check it out.

https://github.com/abpframework/abp/issues/13785

**Review by gizemmutukurt — CHANGES_REQUESTED** [review]
I requested some fixes and changes

File: npm/ng-packs/packages/account/src/lib/guards/extensions.guard.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {Injectable, Injector} from "@angular/core";
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/account/src/lib/guards/extensions.guard.ts]: reformat file

File: npm/ng-packs/packages/core/src/lib/validators/username.validator.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import {AbstractControl, ValidatorFn} from '@angular/forms';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/core/src/lib/validators/username.validator.ts]: reformat file

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-form/extensible-form-prop.component.ts [hunk 3]
```diff
@@ -19,17 +20,17 @@ import {
...
-import { debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
-import { DateAdapter } from '../../adapters/date.adapter';
-import { TimeAdapter } from '../../adapters/time.adapter';
-import { EXTRA_PROPERTIES_KEY } from '../../constants/extra-properties';
-import { ePropType } from '../../enums/props.enum';
-import { FormProp } from '../../models/form-props';
-import { PropData } from '../../models/props';
-import { selfFactory } from '../../utils/factory.util';
-import { addTypeaheadTextSuffix } from '../../utils/typeahead.util';
+import {NgbDateAdapter, NgbTimeAdapter} from '@ng-bootstrap/ng-bootstrap';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-form/extensible-form-prop.component.ts]: reformat file

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-form/extensible-form.component.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import {
   SkipSelf,
   ViewChildren,
 } from '@angular/core';
-import { ControlContainer, FormGroup } from '@angular/forms';
+import {ControlContainer, FormGroup, FormGroupDirective} from '@angular/forms';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/components/extensible-form/extensible-form.component.ts]: reformat file

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/models/form-props.ts [hunk 5]
```diff
@@ -38,6 +38,8 @@ export class FormProp<R = any> extends Prop<R> {
   readonly defaultValue: boolean | number | string | Date;
   readonly options: PropCallback<R, Observable<ABP.Option<any>[]>> | undefined;
   readonly id: string | undefined;
+  readonly template? : Type<any>
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/models/form-props.ts]: reformat file

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/models/props.ts [hunk 6]
```diff
@@ -33,6 +33,8 @@ export abstract class Prop<R = any> {
     public readonly permission: string,
     public readonly visible: PropPredicate<R> = _ => true,
     public readonly isExtra = false,
+    public readonly template? : Type<any>,
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/models/props.ts]: reformat file

File: npm/ng-packs/packages/theme-shared/extensions/src/lib/tokens/extensions.token.ts [hunk 7]
```diff
@@ -3,6 +3,8 @@ import { ActionCallback, ReadonlyActionData as ActionData } from '../models/acti
 import { ExtensionsService } from '../services/extensions.service';
 import { Observable } from 'rxjs';
 import { ePropType } from '../enums/props.enum';
+ import {FormProp} from "../models/form-props";
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/extensions/src/lib/tokens/extensions.token.ts]: reformat file

**Review by gizemmutukurt — APPROVED** [review]

**Pull request merged**